source/TimePicker.js

```diff
@@ -585,14 +625,33 @@
 		meridiemPickerChanged: function (sender, event) {
 			if(this.syncingPickers) return true;
 
-			var hour = this.$.hour.get('value'),
-				value = event.value;
+			var meridiemPicker = event.originator,
+				meridiems = meridiemPicker.get('meridiems'),
+				meridiem = meridiems[meridiemPicker.get('value')],
```

Comment:
Rather than do this, it might make more sense to expose a method on `MeridiemPicker` that returns the current meridiem, or returns a meridiem based on a given value, where this value defaults to the current value.